cmd/cli/app/provider/provider_update.go

```diff
@@ -0,0 +1,509 @@
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/stacklok/minder/internal/util/cli"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+var (
+	errMissingProviderName = errors.New("provider name flag is missing")
+	errMissingProject      = errors.New("project flag is missing")
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Updates a provider's configuration",
+	Long: `The minder provider update command allows a user to update a provider's
+configuration after enrollement.`,
+	RunE: cli.GRPCClientWrapRunE(UpdateProviderCommand),
+}
+
+type parser func(string) (*reflect.Value, error)
+
+func parseBool(val string) (*reflect.Value, error) {
+	v, err := strconv.ParseBool(val)
+	if err != nil {
+		return nil, fmt.Errorf("expected bool, got %s", val)
+	}
+	res := reflect.ValueOf(v)
+	return &res, nil
+}
+
+func parseInt(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseInt(val, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func parseUint(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseUint(val, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func parseFloat(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseFloat(val, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func identity(val string) (*reflect.Value, error) {
+	res := reflect.ValueOf(val)
+	return &res, nil
+}
+
+var (
+	parserMap = map[reflect.Kind]parser{
+		reflect.Bool:    parseBool,
+		reflect.Int32:   parseInt(32),
+		reflect.Int64:   parseInt(64),
+		reflect.Uint32:  parseUint(32),
+		reflect.Uint64:  parseUint(64),
+		reflect.Float32: parseFloat(32),
+		reflect.Float64: parseFloat(64), // double
+		reflect.String:  identity,       // identity
+	}
+)
+
+type configStruct struct {
+	*minderv1.ProviderConfig
+	//nolint:lll
+	GitHub *minderv1.GitHubProviderConfig `json:"github,omitempty" yaml:"github" mapstructure:"github" validate:"required"`
+	//nolint:lll
+	GitHubApp *minderv1.GitHubAppProviderConfig `json:"github_app,omitempty" yaml:"github_app" mapstructure:"github_app" validate:"required"`
+}
+
+// UpdateProviderCommand is the command for enrolling a provider
+//
+//nolint:gocyclo
+func UpdateProviderCommand(
+	ctx context.Context,
+	cmd *cobra.Command,
+	_ []string,
+	conn *grpc.ClientConn,
+) error {
+	// TODO: get rid of provider flag, only use class
+	providerName := viper.GetString("name")
+	if providerName == "" {
+		return cli.MessageAndError(
```

Comment:
I think you can use `updateCmd.MarkFlagRequired("name")` in  `init` and then cobra catches the missing flag for you.